### PR TITLE
Address issues raised by clang-tidy

### PR DESCRIPTION
**Pull request by marbre**

### CODE REVIEW

File: include/emitc/Target/Cpp/CppEmitter.h [hunk 1]
```diff
@@ -33,33 +33,32 @@ template <typename ForwardIterator, typename UnaryFunctor,
           typename NullaryFunctor>
 inline LogicalResult
 interleaveWithError(ForwardIterator begin, ForwardIterator end,
-                    UnaryFunctor each_fn, NullaryFunctor between_fn) {
+                    UnaryFunctor eachFn, NullaryFunctor betweenFn) {
```
Comment (simon-camp) [inline include/emitc/Target/Cpp/CppEmitter.h:36]: This uses the exact same naming as an upstream function, see [here](https://llvm.org/doxygen/namespacellvm.html#ad75c79282c4734018fe1482b12d03d62).

File: lib/Dialect/EmitC/Conversion/MHLORegionOpsToEmitC.cpp [hunk 2]
```diff
@@ -151,7 +151,7 @@ struct ConvertMhloRegionOpsToEmitCPass
     }
 
     OpBuilder builder(op);
-    auto ctx = op.getContext();
+    auto *ctx = op.getContext();
```
Comment (simon-camp) [inline lib/Dialect/EmitC/Conversion/MHLORegionOpsToEmitC.cpp:154]: We will most likely remove auto anyway :)

**Review by simon-camp — CHANGES_REQUESTED** [review]

File: include/emitc/Target/Cpp/CppEmitter.h (hunk 1 above)
Comment (marbre) [reply include/emitc/Target/Cpp/CppEmitter.h:36]: Indeed, but `clang-tidy` complains anyway and resulted in a build failure of https://reviews.llvm.org/D104632.

File: lib/Dialect/EmitC/Conversion/MHLORegionOpsToEmitC.cpp (hunk 2 above)
Comment (marbre) [reply lib/Dialect/EmitC/Conversion/MHLORegionOpsToEmitC.cpp:154]: Yes, it's just an intermediate change. I think it doesn't hurts.

File: include/emitc/Target/Cpp/CppEmitter.h (hunk 1 above)
Comment (simon-camp) [reply include/emitc/Target/Cpp/CppEmitter.h:36]: Okay that's a compelling reason.

**Review by simon-camp — APPROVED** [review]

**Pull request merged**